### PR TITLE
PAUSE-Update Gemfile.lock and Dockerfile for ruby 2.7.6

### PR DESCRIPTION
**Pull request by pengyin-shan**
1) `Gemfile.lock` needs to be updated for better compatibility (especially dependable upgrade) of the DMP Assisant. It can fix current merging issues such as https://github.com/portagenetwork/roadmap/pull/252 cannot be merged due to version incompatibility.
2) Update `Dockerfile.production` and `Dockerfile` to ruby 2.7.6. A `bundle update rake` is needed to avoid a ruby 2.7.0 error

### CODE REVIEW

**Comment by pengyin-shan** [comment]
@pgwillia I need to update `Gemfile.lock` to ruby 2.7.6 since dependabot's PRs need a version bump to pass the test. Could you take a look to see if it makes sense? (you will see both 

Also I noted the Dockerfile needs to have `bundle update rake` to avoid a bug (bundler cannot find rake). I wonder if other apps have seen similar error?

Again pls ignore Rubocop test for now.

File: Gemfile.lock [hunk 1]
```diff
@@ -501,72 +508,79 @@ GEM
...
-    yard (0.9.26)
+    yard (0.9.28)
+      webrick (~> 1.7.0)
     yard-tomdoc (0.7.1)
       tomparse (>= 0.4.0)
       yard
     zaru (0.3.0)
 
 PLATFORMS
-  ruby
```
Comment (pgwillia) [inline Gemfile.lock:569]: Maybe both ruby and arm64-darwin-22 [platforms](https://bundler.io/v2.3/man/bundle-lock.1.html#SUPPORTING-OTHER-PLATFORMS) would be useful?

**Review by pgwillia — APPROVED** [review]
This is okay, I think.  Just wondering about the platform change.

File: Gemfile.lock (hunk 1 above)
Comment (pengyin-shan) [reply Gemfile.lock:569]: Oh sorry I was just doing a rebase and seems ruby was gone in the latest push...I'll fit it.

Comment (pengyin-shan) [reply Gemfile.lock:569]: sorry @pgwillia as I said in stand-up, I'm gonna pause this PR for a while...I found the dependablebot pushes (version upgrades) are overlapping with my upgrade to 3.1.0 (which also has lots of upgrades). 

I'll take a look at what I have now to make sure I'm not going to duplicate PR and create future conflicts.

**Comment by pengyin-shan** [comment]
Covered in 3.1.0 PR

**Pull request closed without merging**